pixy/httpapi.go

```diff
@@ -198,6 +200,83 @@ func (as *HTTPAPIServer) handleConsume(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+// handleGetOffsets is an HTTP request handler for `GET /topic/{topic}/offsets`
+func (as *HTTPAPIServer) handleGetOffsets(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	topic := mux.Vars(r)[ParamTopic]
+	group, err := getGroupParam(r)
+	if err != nil {
+		respondWithJSON(w, http.StatusBadRequest, errorHTTPResponse{err.Error()})
+		return
+	}
+
+	partitionOffsets, err := as.admin.GetGroupOffsets(group, topic)
+	if err != nil {
+		if err == sarama.ErrUnknownTopicOrPartition {
+			respondWithJSON(w, http.StatusNotFound, errorHTTPResponse{"Unknown topic"})
+			return
+		}
+		respondWithJSON(w, http.StatusInternalServerError, errorHTTPResponse{err.Error()})
+		return
+	}
+
+	partitionOffsetView := make([]partitionOffsetView, len(partitionOffsets))
+	for i, po := range partitionOffsets {
+		partitionOffsetView[i].Partition = po.Partition
+		partitionOffsetView[i].Range.Begin = po.Range.Begin
+		partitionOffsetView[i].Range.End = po.Range.End
+		partitionOffsetView[i].Offset = po.Offset
+		partitionOffsetView[i].Metadata = po.Metadata
+	}
+	respondWithJSON(w, http.StatusOK, partitionOffsetView)
+}
+
+// handleGetOffsets is an HTTP request handler for `POST /topic/{topic}/offsets`
+func (as *HTTPAPIServer) handleSetOffsets(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+
+	topic := mux.Vars(r)[ParamTopic]
+	group, err := getGroupParam(r)
+	if err != nil {
+		respondWithJSON(w, http.StatusBadRequest, errorHTTPResponse{err.Error()})
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		errorText := fmt.Sprintf("Failed to read the request: cause=(%v)", err)
+		respondWithJSON(w, http.StatusBadRequest, errorHTTPResponse{errorText})
+		return
+	}
+
+	var partitionOffsetViews []partitionOffsetView
+	if err := json.Unmarshal(body, &partitionOffsetViews); err != nil {
+		errorText := fmt.Sprintf("Failed to parse the request: cause=(%v)", err)
+		respondWithJSON(w, http.StatusBadRequest, errorHTTPResponse{errorText})
+		return
+	}
+
+	partitionOffsets := make([]PartitionOffset, len(partitionOffsetViews))
+	for i, pov := range partitionOffsetViews {
```

Comment:
Because it is impossible to set range for a topic. Basically the JSON partition offset object in `POST /topic/{}/offsets` request can contain **partition**, **offset** and **metadata**. I could have had another struct for that, but then decided to reuse `partitionOffsetView` mostly because I could not come up with a name for yet another partition offset struct :).